File: python/pi_lab/1_blink.py
import RPi.GPIO as GPIO
import time

ledPin = 11

def setup():
    GPIO.setmode(GPIO.BOARD)      # Use physical board numbering.
    GPIO.setup(ledPin, GPIO.OUT)  # Set the ledPin to OUTPUT mode.
    GPIO.output(ledPin, GPIO.LOW) # Make ledPin output LOW level.
    print('Using pin' + str(ledPin))

def loop():
    while True:
        GPIO.output(ledPin, GPIO.HIGH) # Make ledPin output HIGH level to turn on LED.
        print('LED turned on. >>>')
        time.sleep(1)

        GPIO.output(ledPin, GPIO.LOW)  # Turn off LED.
        print('LED turned off. <<<')
        time.sleep(1)

def destroy():
    GPIO.cleanup()                # Release all GPIO.

print('Program is starting... \n')
setup()
try:
    loop()
except KeyboardInterrupt:
    destroy()

File: python/pi_lab/1_blink.c
// Lab 1
// NOTE: Compile with: gcc 1_blink.c -lwiringPi -o 1_blink

#include <wiringPi.h>
#include <stdio.h>
#define ledPin 0 //define the led pin number

void main(void)
{
    printf("Program is starting ... \n");
    wiringPiSetup(); //Initialize wiringPi.
    pinMode(ledPin, OUTPUT);//Set the pin mode
    printf("Using pin%d\n", ledPin); //Output information on terminal

    while (1) {
        digitalWrite(ledPin, HIGH); //Make GPIO output HIGH level
        printf("led turned on >>>\n"); //Output information on terminal
        delay(1000); //Wait for 1 second
        digitalWrite(ledPin, LOW); //Make GPIO output LOW level
        printf("led turned off <<<\n"); //Output information on terminal
        delay(1000); //Wait for 1 second
    }
}
